### PR TITLE
8256283: IndexOutOfBoundsException when sorting a TreeTableView

### PR DESCRIPTION
**Pull request by arapte**
This particular issue JDK-8256283, is a specific case of IOOBE when, rootItem is not shown, some children including first child are selected, then all children are removed and sort() is invoked. The sort() fails with an IOOBE.
This PR only addresses this specific IOOBE.
Root cause of this issue is that the selection is not cleared after rootItems children are removed. In addition to this, there are few other scenarios when selection is not updated correctly, which are collected under an umbrella task [JDK-8248217](https://bugs.openjdk.java.net/browse/JDK-8248217). Fix for [JDK-8248217](https://bugs.openjdk.java.net/browse/JDK-8248217) would require good amount refactoring of selection model.

The fix for this issue is to avoid sort() when rootItem.getChildren().isEmpty().
Added a unit test with the fix, which fails without fix and passes with fix.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8256283](https://bugs.openjdk.java.net/browse/JDK-8256283): IndexOutOfBoundsException when sorting a TreeTableView

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/384/head:pull/384`
`$ git checkout pull/384`

### CODE REVIEW

File: modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java [hunk 1]
```diff
@@ -579,7 +579,7 @@ public static int getNodeLevel(TreeItem<?> node) {
         @Override public Boolean call(TreeTableView table) {
             try {
                 TreeItem rootItem = table.getRoot();
-                if (rootItem == null) return false;
+                if (rootItem == null | rootItem.getChildren().isEmpty()) return false;
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java:582]: That should be `||` (boolean OR). In addition to being less clear, this will get an NPE if `rootItem` is ever null, since the bit-wise `|` operator doesn't short-curcuit the rest of the `if` test if the first term is true.

Given that this would have caused a regression, and that we don't have a test that would catch it, can you add a test that sets the root to null and calls sort? It will fail with this version of the fix, but would pass without the fix and should pass once you update your fix.

**Review by kevinrushforth — CHANGES_REQUESTED** [review]
Comments inline.

File: modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java (hunk 1 above)
Comment (arapte) [reply modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java]: Sorry about that, It was a typo. But it did reveal the missing test. 
Corrected the operator to `||` and added a test as you suggested which fails only with the commit 1 of this PR.

**Review by kevinrushforth — APPROVED** [review]
